### PR TITLE
fix: do not update secret when tls.secretName is empty

### LINKED ISSUE #3711: Ingresses don't get an IP, when `spec.tls[].secretName` is missing - 2.8.x
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Ingresses don't get an IP, when `spec.tls[].secretName` is missing.
No apparent error is logged regarding this.

But when spec.tls[].secretName is defined, the ingress get an IP assigned.

### Expected Behavior

Ingresses do not require `spec.tls[].secretName` to be defined. 

Prior to 2.8.x, it was possible to use the default certificate by defining following to Kong (tested in 2.7.0)
```
env:
  ssl_cert: /etc/secrets/.../tls.crt
  ssl_cert_key: /etc/secrets/.../tls.key
```
And then we didn't need to define `spec.tls[].secretName` in all ingresses.

### Steps To Reproduce

1. Add an ingress with `spec.tls[].hosts[] = host`, like:
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: tls-example-ingress
  annotations:
    kubernetes.io/ingress.class: kong
spec:
  tls:
  - hosts:
      - https-example.foo.com
  rules:
  - host: https-example.foo.com
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: service1
            port:
              number: 80
```
2. See no IP is provided for the ingress
3. Now add `spec.tls[].secretName = secretName`, like
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: tls-example-ingress
  annotations:
    kubernetes.io/ingress.class: kong
spec:
  tls:
  - hosts:
      - https-example.foo.com
    secretName: testsecret-tls # This is new
  rules:
  - host: https-example.foo.com
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: service1
            port:
              number: 80
```
4. See an IP now is provided for the ingress

### Kong Ingress Controller version

```shell
v2.8.x
```

### Kubernetes version

1.24

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

When `ingress.Spec.TLS` does not contain a secret, skip the updating of referred secrets because they are "Optional".

**Which issue this PR fixes**:

fixes #3711 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -63,6 +63,16 @@ Adding a new version? You'll need three changes:
...
  - [0.0.4 and prior](#004-and-prior)
 
+## [2.9.0]
+
+> Release date: TBD
+
+### Fixed
+
+- Fixed the issue that status of ingress is not updated when `secretName` is
+  not spefied in `ingress.spec.tls`.
```
Comment (pmalek) [inline CHANGELOG.md:73]:
```suggestion
- Fixed the issue where the status of an ingress is not updated when `secretName` is
  not specified in `ingress.spec.tls`.
```

**Review by pmalek — COMMENTED** [review]
The conformance tests for Gateway API are failling but I'm not sure if that's related: 🤔 

```
    httproute-observed-generation-bump.go:69: 
        	Error Trace:	/home/runner/go/pkg/mod/sigs.k8s.io/gateway-api@v0.6.1/conformance/tests/httproute-observed-generation-bump.go:69
        	Error:      	Received unexpected error:
        	            	Operation cannot be fulfilled on httproutes.gateway.networking.k8s.io "observed-generation-bump": the object has been modified; please apply your changes to the latest version and try again
        	Test:       	TestGatewayConformance/HTTPRouteObservedGenerationBump/observedGeneration_should_increment
        	Messages:   	error updating the HTTPRoute: Operation cannot be fulfilled on httproutes.gateway.networking.k8s.io "observed-generation-bump": the object has been modified; please apply your changes to the latest version and try again
--- FAIL: TestGatewayConformance/HTTPRouteObservedGenerationBump/observedGeneration_should_increment (0.02s)
```

**Comment by randmonkey** [comment]
> The conformance tests for Gateway API are failling but I'm not sure if that's related: thinking
> 
> ```
>     httproute-observed-generation-bump.go:69: 
>         	Error Trace:	/home/runner/go/pkg/mod/sigs.k8s.io/gateway-api@v0.6.1/conformance/tests/httproute-observed-generation-bump.go:69
>         	Error:      	Received unexpected error:
>         	            	Operation cannot be fulfilled on httproutes.gateway.networking.k8s.io "observed-generation-bump": the object has been modified; please apply your changes to the latest version and try again
>         	Test:       	TestGatewayConformance/HTTPRouteObservedGenerationBump/observedGeneration_should_increment
>         	Messages:   	error updating the HTTPRoute: Operation cannot be fulfilled on httproutes.gateway.networking.k8s.io "observed-generation-bump": the object has been modified; please apply your changes to the latest version and try again
> --- FAIL: TestGatewayConformance/HTTPRouteObservedGenerationBump/observedGeneration_should_increment (0.02s)
> ```

I think this failure is not related to the PR itself, but worth creating an issue.

File: CHANGELOG.md (hunk 1 above)
Comment (randmonkey) [reply CHANGELOG.md]: updated CHANGELOG.

**Review by pmalek — APPROVED** [review]

**Pull request merged**